src/sql/workbench/api/browser/mainThreadConnectionManagement.ts

```diff
@@ -101,22 +102,7 @@ export class MainThreadConnectionManagement extends Disposable implements MainTh
 			return Promise.resolve(undefined);
 		}
 
-		let connection: azdata.connection.ConnectionProfile = {
-			providerId: profile.providerName,
-			connectionId: profile.id,
-			connectionName: profile.connectionName,
-			serverName: profile.serverName,
-			databaseName: profile.databaseName,
-			userName: profile.userName,
-			password: profile.password,
-			authenticationType: profile.authenticationType,
-			savePassword: profile.savePassword,
-			groupFullName: profile.groupFullName,
-			groupId: profile.groupId,
-			saveProfile: profile.savePassword,
-			azureTenantId: profile.azureTenantId,
-			options: profile.options
-		};
+		let connection = convertToRpcConnectionProfile(profile);
```

Comment:
Have you tested this? I'm pretty sure this isn't going to work how you intend since the transfer over the RPC layer lose all the class definitions which means it won't have the getters/setters that look up in the options map. Try calling profile.password (or any other getter) on the extension host side and see what you get. 

It may happen to work for the password change stuff you added since you're only looking at the options field (which is the only thing that's going to populate correctly), but this change is going to break this for anything relying on the top level object properties existing unless I'm missing something here.

Reply:
Changed the convert function to use something akin to the original function (it turns out the options found in the IConnectionProfile wasn't the same as what was in connection.ConnectionProfile, silly me)